File: frontend/src/types/DietPlan.ts
import { User } from '@/types/User';

export interface Meal {
  id: number;
  name: string;
  description?: string;
  mealType: string;
  notes?: string;
}

export interface DietPlanDay {
  id: number;
  dayOfWeek: number;
  meals: Meal[];
}

export interface DietPlanAssignment {
  id: number;
  dietPlanId: number;
  clientId: number;
  client?: User;
  isActive: boolean;
  assignedAt: string;
}

export interface DietPlan {
  id: number;
  name: string;
  trainerId: number;
  trainer?: User;
  days: DietPlanDay[];
  assignments: DietPlanAssignment[];
  createdAt: string;
}

export interface CreateDietPlanRequest {
  name: string;
  days: Omit<DietPlanDay, 'id' | 'meals'> & { meals: Omit<Meal, 'id'>[] }[];
}

export interface UpdateDietPlanRequest {
  name?: string;
  days?: (Omit<DietPlanDay, 'id' | 'meals'> & { meals: Omit<Meal, 'id'>[] })[];
}

export interface AssignDietPlanRequest {
  dietPlanId: number;
  clientIds: number[];
  isActive: boolean;
}

File: backend/Easygym.Domain/Entities/DietPlan.cs
using System.ComponentModel.DataAnnotations;

namespace Easygym.Domain.Entities
{
    public class DietPlan
    {
        public int Id { get; set; }
        [StringLength(100)]
        public required string Name { get; set; }
        public required int TrainerId { get; set; }
        public User? Trainer { get; set; }
        public required List<DietPlanDay> Days { get; set; }
        public List<DietPlanAssignment> Assignments { get; set; } = [];
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}
